lib/puppet/reports/servicenow.rb

```diff
@@ -16,7 +16,17 @@ def process
 
     Puppet.info(sn_log_entry("incident creation conditions: #{incident_creation_conditions}"))
 
-    unless create_incident?(status, resource_statuses, incident_creation_conditions)
+    # Some incident creation conditions depend on the resource events. Thus, we go ahead
+    # and evaluate these 'event' conditions _before_ evaluating the incident creation
+    # conditions so we can iterate through the resource events only once. This results in
+    # cleaner code for a negligible performance hit, where the performance hit occurs if the
+    # incident creation conditions do _not_ contain an 'event' condition.
+    #
+    # Note that event_conditions is a hash of <condition> => <satisfied?>
+    event_conditions = event_conditions(resource_statuses)
+    satisfied_conditions = satisfied_conditions(status, incident_creation_conditions, event_conditions)
```

Comment:
Let's call the `satisfied_conditions` method `calculate_satisfied_conditions` to avoid confusing with the variable name.